examples/Gordo-Reporters-MlFlow.ipynb

```diff
@@ -0,0 +1,353 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using the Gordo Mlflow reporter with AzureML\n",
+    "\n",
+    "## Building on a cluster\n",
+    "When a gordo workflow is generated from a YAML config using `kubectl apply -f config.yml`, the model is built by the model builder pod. If a remote logging \"reporter\" was configured in the `config.yml`, then at the end of the model building step the metadata will be logged with the specified reporter. \n",
+    "\n",
+    "**Note**\n",
+    "When using the MLflow reporter, the cluster running the workflow must have the AzureML workspace credentials set to the environment variable `AZUREML_WORKSPACE_STR`.\n",
+    "\n",
+    "The cluster should use the workspace credentials associated with the deployment stage associated with that cluster, e.g. \"production\", \"staging\", \"testing\", etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from azureml.core.workspace import Workspace\n",
+    "from azureml.core.authentication import InteractiveLoginAuthentication\n",
+    "import mlflow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "config_str = \"\"\"\n",
+    "apiVersion: equinor.com/v1\n",
+    "kind: Gordo\n",
+    "metadata:\n",
+    "  name: test-project\n",
+    "spec:\n",
+    "  deploy-version: 0.48.3\n",
+    "  config:\n",
+    "    machines:\n",
+    "      - dataset:\n",
+    "          tags:\n",
+    "            - Tag1\n",
+    "            - Tag2\n",
+    "          target_tag_list:\n",
+    "            - Tag3\n",
+    "            - Tag4\n",
+    "          train_end_date: '2019-03-01T00:00:00+00:00'\n",
+    "          train_start_date: '2019-01-01T00:00:00+00:00'\n",
+    "          asset: asgb\n",
+    "          data_provider:\n",
+    "            type: RandomDataProvider\n",
```

Comment:
I find it a bit confusing this config. It has "asset:asgb" but tag names that are just examples. And it uses RandomDataProvider.

What about making a config that would work, if you only inserted tags we recognize? Meaning, get rid of the data_provider and the asset:, and fix the reporter according to @milesgranger feedback. And test the config, that it works, also with the mlflow reporter?

Reply:
Ok, the notebook is now updated with an example from the test project, including setting interactive login with the data provider. 

@milesgranger I also updated how the logging methods pull the secrets to make it a tad simpler, just in case you have any thoughts on that.